File: src/front-end/lists/static/js/custom/models/tag.js

/**
 * Model class for Tags
 */
export class Tag
{
    constructor(newTagObject) {
        this.id         = null;
        this.name       = null;
        this.color      = null;
        this.created_on = null;
        this.user_id    = null;


        for (const key in this) {
            if (newTagObject.hasOwnProperty(key)) {
                this[key] = newTagObject[key];
            }
        }
    }
}

File: src/api/api_lists/models/tag.py
"""
**********************************************************************************
Tag model
**********************************************************************************
"""

from dataclasses import dataclass
from uuid import UUID
from datetime import datetime

@dataclass
class Tag:
    id         : UUID    = None
    name       : str     = None
    color      : str     = None
    created_on: datetime = None
    user_id    : UUID    = None






